test/test_http_adapter.py

```diff
@@ -33,40 +30,7 @@
 key = os.path.join(os.path.dirname(__file__), '../resources/test_ssl.key')
 
 
-def _local_server(tls_version: int, port: int) -> Callable:
```

Comment:
I moved this to a separate file, so that it's easier to use it in other tests too.